### PR TITLE
New SLES-15 SP1 supportserver

### PR DESCRIPTION
**Pull request by klwagner**
The currently available openQA supportserver is SLES-12 SP3. This is an old
product. E.g. it was found to be unable to properly mount/umount SLE-15
btrfs: this ability is needed for the upcoming installation test of
a SLES-15 client on a (remote) supportserver disk.

The available documentation/config on supportserver generation does not
really provide much assistance on how to proceed for SLE-15 and newer.
Therefore this PR provides a generator script and a tentative autoyast file
from which a SLES-15 SP1 supportserver disk image may be generated in a
one-time job run.

**FIXME**  The autoyast file specifies *internal* SUSE repositories. At the time of
this PR this is actually inevitable since the needed crucial `tftp` fix [S:M:12899](http://merkur.qam.suse.de/incident/12899/) for [bsc#1153625](https://bugzilla.suse.com/show_bug.cgi?id=1153625)
has not yet been released. In order to later utilize the official repositories the autoyast file
would have to include sections for the needed registrations.

- Related ticket:  https://progress.opensuse.org/issues/50144
- Needles: (none)
### [SLES-15 SP1 supportserver generation job](http://polya.suse.de/tests/617)
Resulting [disk image](http://polya.suse.de/tests/617/asset/hdd/openqa_support_server_sles15sp1.x86_64.qcow2)

### Regression tests: existing testsuites qa{,m}_kernel_multipath_flakyserver

-   [client (SLES-15 SP1)](http://polya.suse.de/tests/699), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/698)
-   [client (SLES-15 GA)](http://polya.suse.de/tests/719), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/718)
-   [client (SLES-12 SP4)](http://polya.suse.de/tests/714), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/713)

### Regression tests: Existing testsuites qa{,m}_kernel_multipath
-  [client (SLES-15 SP1)](http://polya.suse.de/tests/743), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/742)
-  [client (SLES-15 SP1)](http://polya.suse.de/tests/734), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/733)

### Validation tests: upcoming testsuite "install and boot remote multipathed system disk"
See [PR#8984](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/8984).
-  [client (SLES-15 SP1)](http://polya.suse.de/tests/651), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/650)
-  [client (SLES-15 GA)](http://polya.suse.de/tests/645), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/644)
-  [client (SLES-12 SP4)](http://polya.suse.de/tests/657), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/656)
-  [client (SLES-12 SP3 LTSS)](http://polya.suse.de/tests/659), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/658)
-  [client (SLES-12 SP2 LTSS)](http://polya.suse.de/tests/660), [supportserver (SLES-15 SP1) ](http://polya.suse.de/tests/661)

### CODE REVIEW

**Comment by klwagner** [comment]
c3d1516: rebase on current upstream master c73b1740

File: data/supportserver/autoyast_supportserver_x86_sle15.xml [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+        <priority config:type="integer">99</priority>
+        <ask_on_error config:type="boolean">false</ask_on_error>
+        <confirm_license config:type="boolean">false</confirm_license>
+        <ask_user config:type="boolean">false</ask_user>
+        <selected config:type="boolean">true</selected>
+      </listentry>
+
+      <listentry>
+	<!--   NEEDED  not-yet-official update 5.2-5.3.1 for BUGGY tftp-5.2-3.22, bsc#1153625  -->
+        <media_url>http://download.suse.de/ibs/SUSE:/Maintenance:/12899/SUSE_Updates_SLE-Module-Basesystem_15-SP1_x86_64</media_url>
```
Comment (baierjan) [inline data/supportserver/autoyast_supportserver_x86_sle15.xml]: If I am not mistaken, this is no longer valid (empty repository) as the update is already released.

Comment (klwagner) [reply data/supportserver/autoyast_supportserver_x86_sle15.xml]: Right.

It wasn't, though, at the time this supportserver was first needed. The absence of this dr*tted bug would have saved me quite some hunting work. Not a fond memory indeed.

Complied with your request with commit 0019204.

Comment (klwagner) [reply data/supportserver/autoyast_supportserver_x86_sle15.xml]: Could you please mark your change request as addressed? Thanks!

**Comment by klwagner** [comment]
Official verification run after merging: [sle-15-SP1-Server-DVD-x86_64-Build228.2-supportserver_generator@64bit](https://openqa.suse.de/tests/3781166) on `openqa.suse.de`

**Pull request merged**